src/features/tokens/TokenSelectField.tsx

```diff
@@ -26,23 +29,34 @@ export function TokenSelectField({ name, disabled, setIsNft }: Props) {
     const tokensWithRoute = warpCore.getTokensForRoute(origin, destination);
     let newFieldValue: number | undefined;
     let newIsAutomatic: boolean;
+
+    // Use token from persistance
+    if (query.token && typeof query.token === 'string') {
+      // Check if tokenIndex exists
+      const tokenIndex = tryParseAmount(query.token)?.toNumber();
+      if (tokenIndex !== undefined) {
+        const token = warpCore.tokens[tokenIndex];
```

Comment:
The problem with using token index as the query param value is that the index of a token can change between page loads - like when new routes are added or changed. That means someone could get a token that's different than what the link sender expected. Links should be more durable so I recommend you use token address instead and then search for the token index using the address and chains. If the chains aren't set but the token is, ignore it.